lib/web/join_tokens.go

```diff
@@ -52,6 +53,8 @@ type nodeJoinToken struct {
 	Expiry time.Time `json:"expiry,omitempty"`
 	// Method is the join method that the token supports
 	Method types.JoinMethod `json:"method"`
+	// RefResourceID contains the id of the RefResourceID label
+	RefResourceID string `json:"refResourceID,omitempty"`
```

Comment:
JSON tags should be `snake_case`:

```suggestion
	RefResourceID string `json:"ref_resource_id,omitempty"`
```

but mind the other comment about generic labeling instead

Reply:
I think it's ok to return this as a part of labels (that will only contain a single element with `teleport.internal/resource-id` for now), and the UI can search for `teleport.internal/resource-id` label.

Reply:
(@r0mant I think your comment here was meant to this thread https://github.com/gravitational/teleport/pull/15114#discussion_r936427542)

I usually default to snake_case but I recall someone asking me to use camelCase for HTTP responses
@kimlisa I think we talked about this in a previous PR :thinking: Can you please confirm?

Reply:
yes please, it's b/c of JS convention (camelCasing) and removing the need to convert back and forth between camel and snake. but at this point, should we care? we are so big now that newer stuff has snake casing leaking in and using camel casing in the web layer is a knowledge gained from word of mouth